wear/src/main/kotlin/io/homeassistant/companion/android/data/WearDns.kt

```diff
@@ -0,0 +1,111 @@
+package io.homeassistant.companion.android.data
+
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.MessageClient
+import com.google.android.gms.wearable.Wearable
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.time.Instant
```

Comment:
I would be in favor of using `import kotlin.time.Instant` instead and a `kotlin.time.Clock` that you get in the constructor that default to `Clock.System` so you can override it in your test since you are using `now()` to check the cache.

It will also allow you to make this test a bit nicer
```
        if (cached != null && cached.storedAt + 5.minutes > now) {
```